GitCommands/Git/GitModule.cs

```diff
@@ -2470,10 +2469,14 @@ public string GetEffectiveSetting(string setting)
             return EffectiveConfigFile.GetValue(setting);
         }
 
-        public string GetEffectiveGitSetting(string setting, bool cache = true)
+        public EffectiveGitSetting GetEffectiveGitSetting(string setting, bool cache = true)
         {
             GitArgumentBuilder args = new("config") { "--includes", "--get", setting };
-            return GitExecutable.GetOutput(args, cache: cache ? GitCommandCache : null).Trim();
+            ExecutionResult result = GitExecutable.Execute(args, cache: cache ? GitCommandCache : null, throwOnErrorExit: false);
+
+            int resultCode = result.ExitCode ?? 0;
+
+            return (Enum.IsDefined(typeof(GitConfigStatus), resultCode) ? (GitConfigStatus)resultCode : null, result.StandardOutput?.Trim() ?? "");
```

Comment:
You can use [`IsDefined<TEnum>(TEnum)` notation](https://learn.microsoft.com/dotnet/api/system.enum.isdefined?view=net-7.0#system-enum-isdefined-1(-0)).

[nit] Parens are redundant.

Reply:
>You can use [IsDefined<TEnum>(TEnum) notation](https://learn.microsoft.com/dotnet/api/system.enum.isdefined?view=net-7.0#system-enum-isdefined-1(-0)).


Not really 

![image](https://github.com/gitextensions/gitextensions/assets/3629489/a19c2148-d45c-430c-a8c1-93b5fb9fffee)
![image](https://github.com/gitextensions/gitextensions/assets/3629489/fe38187f-0b99-4047-b5f4-ed6e75cd2377)